src/test/groovy/ee/tuleva/onboarding/mandate/email/MandateEmailServiceSpec.groovy

```diff
@@ -3,83 +3,159 @@ package ee.tuleva.onboarding.mandate.email
-        false                | false                        | false    || ["mandate", "pillar_3", "suggest_2"]
-        false                | false                        | true     || ["mandate", "pillar_3", "suggest_2"]
-        true                 | false                        | false    || ["mandate", "pillar_3", "suggest_2"]
-        true                 | false                        | true     || ["mandate", "pillar_3", "suggest_2"]
-        true                 | true                         | false    || ["mandate", "pillar_3", "suggest_member"]
-        true                 | true                         | true     || ["mandate", "pillar_3"]
-    }
+  MandateEmailContentService emailContentService = Mock(MandateEmailContentService)
+  EmailService emailService = Mock(EmailService)
+  def now = Instant.parse("2021-09-01T10:06:01Z")
+  MandateEmailService mandateEmailService = new MandateEmailService(emailService, emailContentService, Clock.fixed(now, UTC))
+
+  def "Send second pillar mandate email"() {
+    given:
+    def user = sampleUser().build()
+    def conversion = notFullyConverted()
+    def contactDetails = contactDetailsFixture()
+    def mandate = sampleMandate()
+    def pillarSuggestion = new PillarSuggestion(3, user, contactDetails, conversion)
+    def recipients = [new Recipient()]
+    def message = new MandrillMessage()
+    def subject = "Pensionifondi avaldus"
+    def html = "html"
+    def tags = ["mandate", "pillar_2", "suggest_3"]
+
+    emailContentService.getSecondPillarHtml(*_) >> html
+    emailService.getRecipients(user) >> recipients
+
+    when:
+    mandateEmailService.sendMandate(user, mandate, pillarSuggestion, contactDetails, Locale.ENGLISH)
+
+    then:
+    1 * emailService.newMandrillMessage(recipients, subject, html, tags, _) >> message
+    1 * emailService.send(user, message)
+  }
+
+  def "Send third pillar payment details email"() {
+    given:
+    def user = sampleUser().build()
+    def conversion = fullyConverted()
+    def contactDetails = contactDetailsFixture()
+    def mandate = sampleMandate()
+    def pillarSuggestion = new PillarSuggestion(2, user, contactDetails, conversion)
+    def recipients = [new Recipient()]
+    def message = new MandrillMessage()
+    def subject = "Important information about your 3rd pillar and a copy of the application"
+    def html = "payment details html"
+    def tags = ["mandate"]
+    def locale = Locale.ENGLISH
+
+    emailContentService.getThirdPillarPaymentDetailsHtml(user, contactDetails.getPensionAccountNumber(), locale) >> html
+    emailService.getRecipients(user) >> recipients
+
+    when:
+    mandateEmailService.sendMandate(user, mandate, pillarSuggestion, contactDetails, locale)
+
+    then:
+    1 * emailService.newMandrillMessage(recipients, subject, html, tags, _) >> message
+    1 * emailService.send(user, message)
+  }
+
+  def "Send third pillar suggest second pillar email"() {
```

Comment:
User `@Unroll` for Data Driven Tests

Reply:
From Spock 2, all tests are unrolled by default so less boilerplate!